main.tf

```diff
@@ -17,11 +17,11 @@ resource "google_kms_key_ring" "key_ring" {
 
 resource "google_kms_crypto_key" "symmetric_keys" {
   # see https://cloud.google.com/kms/docs/encrypt-decrypt
-  count           = length(var.symmetric_keys)
-  name            = "sym-${var.symmetric_keys[count.index].key_name}-${var.name_suffix}"
+  for_each        = { for obj in var.symmetric_keys : obj.key_name => obj }
+  name            = "sym-${each.value.key_name}-${var.name_suffix}"
   key_ring        = google_kms_key_ring.key_ring.self_link
   purpose         = "ENCRYPT_DECRYPT"
-  rotation_period = var.symmetric_keys[count.index].rotation_period
+  rotation_period = each.value.rotation_period
   version_template {
     algorithm        = "GOOGLE_SYMMETRIC_ENCRYPTION" # see https://cloud.google.com/kms/docs/reference/rest/v1/CryptoKeyVersionAlgorithm
```

Comment:
@syedrakib comment like this.

Reply:
# see https://cloud.google.com/kms/docs/reference/rest/v1/CryptoKeyVersionAlgorithm

Reply:
``# see https://cloud.google.com/kms/docs/reference/rest/v1/CryptoKeyVersionAlgorithm``

Reply:
When we google, there are tonnes of information available about "GOOGLE_SYMMETRIC_ENCRYPTION" - but finding the **particular page/article** with the **right context** that is **relevant** to this **particular block** may take a few more clicks and surfs and searches. 

Easier when it's right here as a comment. So if anyone developer intends to revise/learn something about why we did what here,, it will be quicker & easier for him/her to find the **exact relevant article** with **relevant context** quickly.

This has helped us shave off several clicks & surfs & minutes of google searches in the past :)